reader_concurrency_semaphore.cc

```diff
@@ -796,6 +796,9 @@ reader_concurrency_semaphore::~reader_concurrency_semaphore() {
 reader_concurrency_semaphore::inactive_read_handle reader_concurrency_semaphore::register_inactive_read(flat_mutation_reader_v2 reader) noexcept {
     auto& permit_impl = *reader.permit()._impl;
     permit_impl.on_register_as_inactive();
+    if (_blessed_permit == &permit_impl) {
+        _blessed_permit = nullptr;
+    }
```

Comment:
Do we need to kick off waiters? They may have been blocked due to the blessed permit monopolizing memory.

Reply:
If there are waiters, the read is going to be evicted and we already kick off waiters for that case.
And now, one of those waiters will be able to become blessed.

Reply:
I see. Please add a comment then. The waking strategy is fragile (necessarily so, since spurious wakeups eat cycles).

Reply:
done in v2